plotly/plotly/plotly.py

```diff
@@ -1449,6 +1454,81 @@ def _send_to_plotly(figure, **plot_options):
     return r
 
 
+def get_grid(grid_url, raw=False):
+    """
+    Returns a JSON figure representation for the specified grid.
+
+    :param (bool) raw: if False, will output a Grid instance of the JSON grid
+    being retrieved. If True, raw JSON will be returned.
+    """
+    credentials = get_credentials()
+    validate_credentials(credentials)
+    username, api_key = credentials['username'], credentials['api_key']
+    headers = {'plotly-username': username,
+               'plotly-apikey': api_key,
+               'plotly-version': version.__version__,
+               'plotly-platform': 'python'}
+    upload_url = _api_v2.api_url('grids')
+
+    # extract path in grid url
+    url_path = six.moves.urllib.parse.urlparse(grid_url)[2][2:]
+    if url_path[-1] == '/':
+        url_path = url_path[0: -1]
+    url_path = url_path.replace('/', ':')
+
+    meta_get_url = upload_url + '/' + url_path
+    get_url = meta_get_url + '/content'
+
+    r = requests.get(get_url, headers=headers)
+    json_res = json.loads(r.text)
+
+    # make request to grab the grid id (fid)
+    r_meta = requests.get(meta_get_url, headers=headers)
+    json_res_meta = json.loads(r_meta.text)
+    retrieved_grid_id = json_res_meta['fid']
+
+    if raw is False:
+        return Grid(json_res, retrieved_grid_id)
+    else:
+        return json_res
+
+
+def create_animations(figure, filename=None, sharing='public'):
+    """
+    Put description here.
+
+    For parameter descriptions, see the doc string for `py.plot()`.
+    `private` is not supported currently for param 'sharing'. Returns the
+    url for the plot if response is ok.
+    """
+    credentials = get_credentials()
+    validate_credentials(credentials)
+    username, api_key = credentials['username'], credentials['api_key']
+    auth = HTTPBasicAuth(str(username), str(api_key))
+    headers = {'Plotly-Client-Platform': 'python'}
+
+    json = {
+        'figure': figure,
+        'world_readable': 'true'
+    }
+
+    # set filename if specified
+    if filename:
+        json['filename'] = filename
+
+    # set sharing
+    if sharing == 'public':
+        json['world_readable'] = 'true'
+    elif sharing == 'private':
+        json['world_readable'] = 'false'
+
+    api_url = _api_v2.api_url('plots')
+    r = requests.post(api_url, auth=auth, headers=headers, json=json)
+
+    json_r = json.loads(r.text)
+    return json_r['file']['web_url']
```

Comment:
I forget if this URL includes the title text at the end of the URL (e.g. `https://plot.ly/~chris/15/my-title-of-my-graph`) or not. We should make sure that we're returning the minimal version of the url like `https://plot.ly/~chris/15` just like how `py.plot` does